### DISCUSSION TITLE
SBFspot stopped uploading to PVOutput

### DISCUSSION THREAD

**Discussion by svendt** [SBFspot/SBFspot#685] [Q&A]
Hello,

Since a couple of days SBFspot is not uploading to PVOutput anymore.
2 inverters SMA SB-4500TL + TL-2500TL

Config should be ok because:
- nothing changed to config or to PVOutput settings
- manual uploading via command "/usr/local/bin/sbfspot.3/SBFspot -ad62 -am2 -ae2 -v -finq -nocsv" works fine and all missing input is uploaded to PVOutput without problems
- time is ok according "/usr/local/bin/sbfspot.3/SBFspot -v -nosql -nocsv -finq | grep -I time"
- BT is ok according "hcitool scan"

Is not working when manually invoked (no output, script "keeps running", no error message:
/usr/local/bin/sbfspot.3/SBFspotUploadDaemon -c /usr/local/bin/sbfspot.3/SBFspotUpload.cfg

/var/log/syslog is not revealing more info

1) What can I do or check to further troubleshoot?
2) Worst case, can I put for instance "/usr/local/bin/sbfspot.3/SBFspot -ad1 -am1 -ae1 -v -finq -nocsv" in a 5 minute crontab or will this invoke problems to SBFspot or PVOutput?

thanks!

**Comment by sbf-** [comment]
> 2\. Worst case, can I put for instance "/usr/local/bin/sbfspot.3/SBFspot -ad1 -am1 -ae1 -v -finq -nocsv" in a 5 minute crontab

SBFspot has to run every 5 min so yes, you should have a crontab configured.
But it worked before so there is/was a crontab?

See also [No uploads to PVOutput.org](https://github.com/SBFspot/SBFspot/wiki/No-uploads-to-PVOutput.org)

**Reply by svendt** [reply]
Yes, at this moment the crontab is:
```
## SBFspot
*/5 6-22 * * * /usr/local/bin/sbfspot.3/daydata
55 05 * * * /usr/local/bin/sbfspot.3/monthdata
0 1 * * * find /var/log/sbfspot.3/ -name "*.log" -mtime +7 -delete
```
When I run the first one, I get an error about the log files (that doesn't make sense I think since everything is in the db?):
`/usr/local/bin/sbfspot.3/daydata: line 4: /var/log/sbfspot.3/V66_20240115.log: No such file or directory`

AND it worked previously and I haven't touched the crontab.
The content of the script is:
```
#!/bin/bash
#
log=/var/log/sbfspot.3/V66_$(date '+%Y%m%d').log
/usr/local/bin/sbfspot.3/SBFspot -v -ad1 -am0 -ae1 -cfg/usr/local/bin/sbfspot.3/SBFspot.cfg $1 $2 $3 $4 $5 &>>$log
```
(haven't tried to remove the log reference to see if it gets better, but again, worked previously without problem)
(could replace it with "/usr/local/bin/sbfspot.3/SBFspot -v -ad1 -am0 -ae1 -cfg/usr/local/bin/sbfspot.3/SBFspot.cfg" tomorrow when there is sun to test, but is more or less what I added at this moment to the cron as a workaround, so in fact both should work than ("/usr/local/bin/sbfspot.3/SBFspot -ad1 -am1 -ae1 -v -finq -nocsv")

I see now that daydata and monthdata are owned by root.root and not my user, could that be the issue? R-acces is ok
```
-rwxr-xr-x 1 root     root        178 Sep 13 20:44 daydata
-rwxr-xr-x 1 root     root        187 Sep 13 20:44 monthdata
```

**Reply by sbf-** [reply]
Does the directory `/var/log/sbfspot.3` exist?

root.root owner is OK

**Reply by svendt** [reply]
Thanks for your feedback!
Directory was not existing, have created it, but no solution.

The behaviour is odd, see below for current crontab.
1) daydata command (commented for the moment) is not working, content script:
```
#!/bin/bash
#
log=/var/log/sbfspot.3/V66_$(date '+%Y%m%d').log
/usr/local/bin/sbfspot.3/SBFspot -v -ad1 -am0 -ae1 -cfg/usr/local/bin/sbfspot.3/SBFspot.cfg $1 $2 $3 $4 $5 &>>$log
```
But `/usr/local/bin/sbfspot.3/SBFspot -ad1 -am1 -ae1 -v -finq -nocsv` works fine.
```
## SBFspot
#*/5 5-22 * * * /usr/local/bin/sbfspot.3/daydata
55 05 * * * /usr/local/bin/sbfspot.3/monthdata
0 1 * * * find /var/log/sbfspot.3/ -name "*.log" -mtime +7 -delete
*/5 5-22 * * * /usr/local/bin/sbfspot.3/SBFspot -ad1 -am1 -ae1 -v -finq -nocsv
```
Summarised: Crontab scheduling is OK, default command does not work, workaround is working fine.
So no big issue anymore, but very odd at least :)

**Reply by sbf-** [reply]
The only difference I see is the `-nocsv` argument to disable CSV export. If the daydata script tries to export CSV files to a non existing directory it might fail.

**Reply by svendt** [reply]
No, directory exists (*.db, *.db-shm, *.db-wal files are there).

If I put the working command (/usr/local/bin/sbfspot.3/SBFspot -ad1 -am1 -ae1 -v -finq -nocsv) in the /usr/local/bin/sbfspot.3/daydata script (replacing the default command), the working command is not working anymore, so it seems that launching the daydata is the issue somehow.
